packages/hub/searchers.js

```diff
@@ -161,14 +161,15 @@ class Searchers {
     if (result) {
       let schema = await schemaPromise;
       let includePaths = (get(query, 'include') || '').split(',');
-      let pristineResult = await (this.createDocumentContext({
+      let documentContext = this.createDocumentContext({
         branch,
         schema,
         includePaths,
         upstreamDoc: result,
-      }).pristineDoc());
+      });
+      let pristineResult = await documentContext.pristineDoc();
 
-      let authorizedResult = await schema.applyReadAuthorization(pristineResult, { session });
+      let authorizedResult = await documentContext.applyReadAuthorization(pristineResult, { session });
```

Comment:
@balinterdi we should not have the ability to apply read authorization against any document--rather applying read authorization for a context should just apply it against it's own document. it seems like this pattern that we have can be easily misused by using document contexts to apply read authorization against documents that have nothing to do with the contexts

Reply:
Maybe, but do you think the behavior has changed between the old and new version? The main difference I see is that we now call `applyReadAuthorization` on the document context (which we also had in the previous version) whereas we now call it on the DC itself.

Reply:
well the purpose of DC is a bit different. it represents saved work for a specific document. I dont think we are using it wrong now, but the API makes it really easy to abuse this.